--- a/src/Abstract/Header.cpp
+++ b/src/Abstract/Header.cpp
@@ -16,9 +16,15 @@
 #include "LIEF/Visitor.hpp"
 #include "LIEF/Abstract/Header.hpp"
 
+#if defined(LIEF_ELF_SUPPORT)
 #include "LIEF/ELF/Binary.hpp"
+#endif
+#if defined(LIEF_MACHO_SUPPORT)
 #include "LIEF/MachO/Binary.hpp"
+#endif
+#if defined(LIEF_PE_SUPPORT)
 #include "LIEF/PE/Binary.hpp"
+#endif
 
 #include <spdlog/fmt/fmt.h>
 
@@ -37,6 +43,7 @@ static constexpr auto ARRAY_MODES = {
   Header::MODES::BITS_64, Header::MODES::THUMB,
 };
 
+#if defined(LIEF_ELF_SUPPORT)
 Header Header::from(const ELF::Binary& elf) {
   if constexpr (!lief_elf_support) {
     return {};
@@ -131,7 +138,9 @@ Header Header::from(const ELF::Binary& elf) {
 
   return hdr;
 }
+#endif
 
+#if defined(LIEF_PE_SUPPORT)
 Header Header::from(const PE::Binary& pe) {
   if constexpr (!lief_pe_support) {
     return {};
@@ -195,7 +204,9 @@ Header Header::from(const PE::Binary& pe) {
   }
   return hdr;
 }
+#endif
 
+#if defined(LIEF_MACHO_SUPPORT)
 Header Header::from(const MachO::Binary& macho) {
   if constexpr (!lief_macho_support) {
     return {};
@@ -290,6 +301,7 @@ Header Header::from(const MachO::Binary& macho) {
 
   return hdr;
 }
+#endif
 
 std::vector<Header::MODES> Header::modes_list() const {
   std::vector<MODES> flags;
--- a/src/asm/asm.cpp
+++ b/src/asm/asm.cpp
@@ -49,7 +49,9 @@
 #include "LIEF/asm/powerpc/registers.hpp"
 
 #include "LIEF/Abstract/Binary.hpp"
+#if defined(LIEF_COFF_SUPPORT)
 #include "LIEF/COFF/Binary.hpp"
+#endif
 
 #include "internal_utils.hpp"
 #include "messages.hpp"
@@ -104,6 +106,7 @@ assembly::Engine* Binary::get_engine(uint64_t) const {
 }
 
 
+#if defined(LIEF_COFF_SUPPORT)
 namespace COFF {
 assembly::Engine* Binary::get_engine(uint64_t) const {
   return nullptr;
@@ -126,6 +129,7 @@ Binary::instructions_it Binary::disassemble(const uint8_t* /*buffer*/, size_t /*
   return make_empty_iterator<assembly::Instruction>();
 }
 }
+#endif
 
 namespace assembly {
 namespace details {
