--- a/include/LIEF/COFF/Binary.hpp
+++ b/include/LIEF/COFF/Binary.hpp
@@ -225,7 +225,6 @@ class LIEF_API Binary {
   std::string to_string() const;
 
   LIEF_API friend std::ostream& operator<<(std::ostream& os, const Binary& bin) {
-    os << bin.to_string();
     return os;
   }

--- a/include/LIEF/COFF/Symbol.hpp
+++ b/include/LIEF/COFF/Symbol.hpp
@@ -259,7 +259,6 @@ class LIEF_API Symbol : public LIEF::Symbol {
   LIEF_API friend
     std::ostream& operator<<(std::ostream& os, const Symbol& entry)
   {
-    os << entry.to_string();
     return os;
   }

--- a/include/LIEF/PE/ResourcesManager.hpp
+++ b/include/LIEF/PE/ResourcesManager.hpp
@@ -92,7 +92,6 @@ class LIEF_API ResourcesManager : public Object {
     friend LIEF_API
       std::ostream& operator<<(std::ostream& os, const string_entry_t& str)
     {
-      os << std::to_string(str.id) << ", " << str.string_u8();
       return os;
     }
   };
--- a/include/LIEF/PE/resources/ResourceStringTable.hpp
+++ b/include/LIEF/PE/resources/ResourceStringTable.hpp
@@ -54,7 +54,7 @@ class LIEF_API ResourceStringTable : public Object {
     }
 
     std::string to_string() const {
-      return key_u8() + ": " + value_u8();
+      return "";
     }
 
     friend LIEF_API
--- a/include/LIEF/PE/resources/ResourceVersion.hpp
+++ b/include/LIEF/PE/resources/ResourceVersion.hpp
@@ -276,7 +276,6 @@ class LIEF_API ResourceVersion : public Object {
     LIEF_API friend
       std::ostream& operator<<(std::ostream& os, const fixed_file_info_t& info)
     {
-      os << info.to_string();
       return os;
     }
   };
